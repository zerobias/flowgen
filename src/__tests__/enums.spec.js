import { compiler, beautify } from "..";

it("should handle basic enums", () => {
  const ts = `enum Label {
    LABEL_OPTIONAL,
    LABEL_REQUIRED,
    LABEL_REPEATED,
  }`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});

it("should handle number enums", () => {
  const ts = `enum Label {
    LABEL_OPTIONAL = 1,
    LABEL_REQUIRED = 2,
    LABEL_REPEATED = 3,
  }`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});

it("should handle string enums", () => {
  const ts = `enum Label {
    LABEL_OPTIONAL = 'LABEL_OPTIONAL',
    LABEL_REQUIRED = 'LABEL_REQUIRED',
    LABEL_REPEATED = 'LABEL_REPEATED',
  }`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});
