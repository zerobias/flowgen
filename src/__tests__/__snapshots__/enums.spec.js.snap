// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle basic enums 1`] = `
"declare class Label {
  constructor(...args: empty): mixed;
  static LABEL_OPTIONAL: Label; // 0
  static LABEL_REQUIRED: Label; // 1
  static LABEL_REPEATED: Label; // 2
}
"
`;

exports[`should handle number enums 1`] = `
"declare class Label {
  constructor(...args: empty): mixed;
  static LABEL_OPTIONAL: Label; // 1
  static LABEL_REQUIRED: Label; // 2
  static LABEL_REPEATED: Label; // 3
}
"
`;

exports[`should handle string enums 1`] = `
"declare class Label {
  constructor(...args: empty): mixed;
  static LABEL_OPTIONAL: Label; // LABEL_OPTIONAL
  static LABEL_REQUIRED: Label; // LABEL_REQUIRED
  static LABEL_REPEATED: Label; // LABEL_REPEATED
}
"
`;
